src/content/blog/astro-460.mdx

```diff
@@ -0,0 +1,116 @@
+
+Astro will still work on these older versions of Node until Astro 5, however you may see a warning when installing Astro and certain features may not work as expected.
+
+## Manual routing strategy for internationalization
+
+Astro 4.6 introduces a new manual routing strategy for internationalization. This new strategy allows you to take total control over the routing of your internationalized Astro website, for cases where the default routing strategy doesn't quite meet your needs.
+
+To enable manual routing, set the `i18n.strategy` option in your `astro.config.mjs` to `manual`:
+
+```js
+// astro.config.mjs
+import { defineConfig } from "astro/config";
+
+export default defineConfig({
+  i18n: {
+    locales: ["en", "fr"],
+    defaultLocale: "fr",
+    routing: "manual",
+  },
+});
+```
+
+and then [add a middleware](https://docs.astro.build/en/guides/middleware/) to your project to handle the routing:
+
+```ts
+// src/middleware.ts
+import { redirectToDefaultLocale } from "astro:i18n";
+
+// Example middleware that redirects all requests to the default locale apart for the /about page
+export const onRequest = defineMiddleware(async (context, next) => {
+  if (context.url.startsWith("/about")) {
+    return next();
+  } else {
+    return redirectToDefaultLocale(context, 302);
+  }
+});
+```
+
+Check out the [internationalization documentation](https://docs.astro.build/en/guides/internationalization/) for more information on how to use manual routing with internationalization.
+
+## Ability to move the dev toolbar
+
+Astro 4.6 introduces the ability to move the dev toolbar to different positions around the screen. This can be useful if you have a sticky header that covers the bottom of the screen, or if you just prefer the toolbar to be somewhere else. Thanks to [Ming-jun Lu](https://github.com/mingjunlu) for contributing this feature!
+
+## Experimental: Support for CSRF protection
+
+Astro 4.6 adds experimental partial support for [CSRF protection](https://developer.mozilla.org/en-US/docs/Glossary/CSRF). This feature is currently under an experimental flag and may change in future releases. To enable it, set the `experimental.csrfProtection` and `security.csrfProtection.origin` option in your `astro.config.mjs`:
+
+```js
+// astro.config.mjs
+import { defineConfig } from "astro/config";
+
+export default defineConfig({
+  security: {
+    csrfProtection: {
+      origin: true,
+    },
+  },
+  experimental: {
+    csrfProtection: true,
+  },
+});
+```
+
+Note that this feature is only available for pages that are rendered on-demand (also known as Server Side Rendering).
+
+## Cookies improvements
+
+Thanks to [Farzard](https://github.com/fshafiee), Astro's Cookie helper for deleting cookies (`Astro.cookies.delete`) now allowing setting all the standard cookie attributes instead of just the `path` and `domain` attributes. Perhaps small change, but definitely a useful one!
```

Comment:
```suggestion
Thanks to [Farzard](https://github.com/fshafiee), Astro's helper for deleting cookies (`Astro.cookies.delete`) now allowing setting even more cookie attributes instead of just the `path` and `domain` attributes. Perhaps a "bite-sized" change, but definitely one you can sink your teeth into!
```
Once again, Sarah tries to be fun.

But IMPORTANT, this does only adds certain, explicit attributes, not "all of them" necessarily. So better to keep the description scoped.